gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskImageConfiguration.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.services.compute.model.Image;
+import com.google.common.base.MoreObjects;
+
+import java.util.Objects;
+
+/**
+ * A Google Compute Engine disk image configuration. This class can be used to create images from an
+ * existing Google Compute Engine disk.
+ */
+public class DiskImageConfiguration extends ImageConfiguration {
+
+  private static final long serialVersionUID = 2716403667042981170L;
+
+  private final DiskId sourceDisk;
+  private final String sourceDiskId;
+
+  /**
+   * A builder for {@code DiskImageConfiguration} objects.
+   */
+  public static final class Builder
+      extends ImageConfiguration.Builder<DiskImageConfiguration, Builder> {
+
+    private DiskId sourceDisk;
+    private String sourceDiskId;
+
+    private Builder() {}
+
+    private Builder(DiskImageConfiguration imageConfiguration) {
+      super(imageConfiguration);
+      this.sourceDisk = imageConfiguration.sourceDisk;
+      this.sourceDiskId = imageConfiguration.sourceDiskId;
+    }
+
+    private Builder(Image imagePb) {
+      super(imagePb);
+      this.sourceDisk = DiskId.fromUrl(imagePb.getSourceDisk());
+      this.sourceDiskId = imagePb.getSourceDiskId();
+    }
+
+    /**
+     * Sets the identity of the source disk used to create the image.
+     */
+    public Builder sourceDisk(DiskId sourceDisk) {
+      this.sourceDisk = checkNotNull(sourceDisk);
+      return this;
+    }
+
+    Builder sourceDiskId(String sourceDiskId) {
+      this.sourceDiskId = sourceDiskId;
+      return this;
+    }
+
+    /**
+     * Creates a {@code DiskImageConfiguration} object.
+     */
+    @Override
+    public DiskImageConfiguration build() {
+      return new DiskImageConfiguration(this);
+    }
+  }
+
+  private DiskImageConfiguration(Builder builder) {
+    super(builder);
+    this.sourceDisk = checkNotNull(builder.sourceDisk);
+    this.sourceDiskId = builder.sourceDiskId;
+  }
+
+  /**
+   * Returns the identity of the source disk used to create this image.
+   */
+  public DiskId sourceDisk() {
+    return sourceDisk;
+  }
+
+  /**
+   * Returns the id of the disk used to create this image. This value may be used to determine
+   * whether the image was taken from the current or a previous instance of a given disk name.
+   */
+  public String sourceDiskId() {
```

Comment:
I see, I think @mziccard uses `[class]Id()` and `id()` in [class].java, but anywhere there's a field that points to the identity of another class, he uses `[class]` instead of `[classId]`, probably for the sake of brevity.  In this case, what @mziccard is consistent with other usage.

This again makes me wonder whether we can come up with an alternative to better differentiate the two fields.  Perhaps instead of calling the string id `id`, maybe we could say something bit more verbose to differentiate it, like "serviceGeneratedId" (maybe not the best name, but I couldn't think of a better one after a couple minutes of thought).  The string id used less often in our code, and our users probably shouldn't be depending on it all that much, so more verbosity in my mind is okay.